cloud/services/compute/networks/reconcile.go

```diff
@@ -102,6 +108,11 @@ func (s *Service) createOrGetNetwork(ctx context.Context) (*compute.Network, err
 			return nil, err
 		}
 
+		if s.scope.IsSharedVpc() {
+			log.Error(err, "Shared VPC is enabled. Error looking for network", "name", s.scope.NetworkName())
+			return nil, err
+		}
```

Comment:
This confused me a bit, but I think I get it now. This is essentially a validation. If using shared vpc we need existing subnets. Perhaps an error message such as:

`"Shared VPC is enabled, but could not find find existing network", "name", s.scope.NetworkName())